### DISCUSSION TITLE
Implementing a trivial Creep Model

### DISCUSSION THREAD

**Discussion by batodon** [idaholab/moose#22131] [Q&A Modules: Solid mechanics] [answered]
Hello 

Can anyone help me with this basic Creep model? I’m essentially deriving from the RadialReturnCreepStressUpdateBase class and overriding the “computeResidualInternal” and “ComputeDerivative” virtual functions. I did everything right, but my simulation returns zero values for the creep_strain. Thank you.

![problem](https://user-images.githubusercontent.com/113298585/190883688-2cdc60d5-af17-4cb9-bdc8-b0e33be549d0.PNG)
![solution](https://user-images.githubusercontent.com/113298585/190883765-05c13007-62ae-4b84-9a89-65c23acbe2b5.PNG)
[creep.zip](https://github.com/idaholab/moose/files/9592976/creep.zip)

**Comment by RECHOA** [comment, marked as answer]
In your source file, the `etaMax` is  related to `scalar` ：

>   const ScalarType stress_delta = effective_trial_stress - (_three_shear_modulus * scalar);
>   const ScalarType etaMax = _etaM0 * std::exp(_mvM * stress_delta);

but in `computeDerivative` , when calculating the `creep_rate_derivative`, you didn't consider the derivation of `etaMax `with respect to `scalar`  :
>   const GenericReal<is_ad> creep_rate_derivative =
>       -(std::sqrt(2/3)) *   1/(2*etaMax)* _three_shear_modulus;
